### ISSUE TITLE
Rename main module

### ISSUE THREAD

**Issue by zverok** [Arafatk/tensorflow.rb#21]
I believe it should be `TensorFlow` (with corresponding gem change to `tensor_flow`). Current version is not looking too reasonable.

**Comment by geoffreylitt** [comment]
I would agree with changing capitalization to TensorFlow. That's how it's written on the [website](https://www.tensorflow.org/), so I think that's least surprising to a new developer trying to remember usage.

However, given that the Python module is called `tensorflow`, I would personally support keeping the gem name like that. I also think `tensorflow` is more attractive than `tensor_flow`.

**Comment by zverok** [comment]
@geoffreylitt In fact, nobody will panish us for any solution, but in Ruby there is pretty strong agreement to have correspondence between gem name, main file name, and main gem namespace, so it is expected to either be `gem tensorflow`/`require 'tensorflow'`/`module Tensorflow` or `gem tensor_flow`, `require 'tensor_flow`/`module TensorFlow`.

**Comment by chrhansen** [comment]
If you ask me I'd also prefer the most simple, lowercase one word, `tensorflow`, and loose the `.rb` ending. Also, in general, I think we should stay as close to the naming and API of the python module as possible, unless it conflicts with Ruby conventions. 

I would _prefer_ to see `module TensorFlow` (T and F capitalized), but would rather stick with `require 'tensorflow'` and `module Tensorflow` to follow ruby conventions.

**Comment by zverok** [comment]
OK, then let's make the most reasonable thing: `tensorflow` gem name/main file name and `TensorFlow` module name. I agree all other options are quite ugly.

**Comment by nethsix** [comment]
I prefer `tensorflow`, just to be consistent with the python module, and `Tensorflow` just to be consisten with `tensorflow` (otherwise there may be confusion).

**Comment by jtoy** [comment]
agree tensorflow is best
